ra-optimisation/rao-api/src/main/java/com/farao_community/farao/rao_api/Rao.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.rao_api;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.ra_optimisation.RaoComputationResult;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.powsybl.commons.Versionable;
+import com.powsybl.commons.config.PlatformConfig;
+import com.powsybl.commons.util.ServiceLoaderCache;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.computation.local.LocalComputationManager;
+import com.powsybl.iidm.network.Network;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * RA optimisation main API. It is a utility class (so with only static methods) used as an entry point for running
+ * a Ra optimisation allowing to choose either a specific find implementation or just to rely on default one.
+ *
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-france.com>}
+ */
+public final class Rao {
+
+    private Rao() {
+        throw new AssertionError("Utility class should not been instantiated");
+    }
+
+    private static final Supplier<List<RaoProvider>> PROVIDERS_SUPPLIERS
+            = Suppliers.memoize(() -> new ServiceLoaderCache<>(RaoProvider.class).getServices());
```

Comment:
Why is it necessary to use a Supplier and not directly a List<RaoProvider> whose name would be SUPPLIERS ?

Reply:
A supplier is needed to be sure RaoProvider classes has been provided before getting the list. If you use the List directly, as it is static it will be constructed at class creation by the JVM (potentially before RaoProvider implemntations)